tests/test_app.py

```diff
@@ -0,0 +1,74 @@
+#include "Python.h"
+import os
+import sys
+import tempfile
+from paramiko import SSHClient as ssh
+from paramiko.sftp_client import SFTPClient as sftp
+from io import StringIO
+import paramiko
+import flask_unittest
+from unittest.mock import MagicMock, patch
+import flask.globals
+from flask.testing import FlaskClient
+from app import build_app
+
+
+
+
+class MockSSHClient(object):
+    def __init__(self):
+        sys.stdout.channel = MagicMock()
+        sys.stdout.read = MagicMock(return_value = b"""Mock Output""")
+        sys.stderr.read = MagicMock(return_value = b"""Mock Error""")
+        sftp.file = MagicMock(spec = StringIO)
+
+    def set_missing_host_key_policy(self, policy):
+        pass
+
+    # assume the connection is made
+    def connect(self, ip, port=22, username=None, password=None, key_filename=None):
+        pass
+
+    # return local sys.std streams
+    def exec_command(self, command):
+        return (sys.stdin, sys.stdout, sys.stderr)
+
+    def open_sftp(self):
+        return MagicMock(spec = sftp)
+
+    def close(self):
+        pass
+
+
+
+class TestApp(flask_unittest.ClientTestCase):
+    app = build_app()
+
+    def setUp(self, client):
+        self.assertTrue(client is not None)
+        self.assertTrue(isinstance(client, FlaskClient))
+
+    def tearDown(self, client):
+        self.assertTrue(client is not None)
+        self.assertTrue(isinstance(client, FlaskClient))
+
+    def test_setup(self, client):
+        pass
+
+    def test_index_page(self, client):
+        rv: Response = client.get('/')
```

Comment:
instead of abbreviating (rv), can you spell out for better readability. Change here and everywhere else.  Otherwise looks good.